lib/fluent/plugin/buffer/file_chunk.rb

```diff
@@ -219,13 +219,17 @@ def restore_metadata(bindata)
             # old type of restore
             data = Fluent::MessagePackFactory.msgpack_unpacker(symbolize_keys: true).feed(bindata).read rescue {}
           end
+          raise FileChunkError, "invalid meta data" if data.nil? || !data.is_a?(Hash)
+          raise FileChunkError, "invalid unique_id" unless data[:id]
+          raise FileChunkError, "invalid created_at" unless data[:c].to_i > 0
+          raise FileChunkError, "invalid modified_at" unless data[:m].to_i > 0
```

Comment:
`restore_metadata` has been invoked only when `*.meta` file is existed.
This means that `write_metadata` has been invoked before, and the meta file was created.

`write_metadata checks that the file is not corrupted by ensuring that the field to be written has the appropriate value.

In here,  it checks the field to be written by `write_metadata` has the appropriate value.
If the value is not appropriate, it is likely that some kind of error occurred during writing.

Reply:
Could you please make it clear that this does not cause any compatibility issues with the v0.14 API in https://github.com/fluent/fluentd/pull/4998#issue-3129583759?

Reply:
Thanks!

```diff
- However, it causes error with default value.
+ However, it can miss the corruption and result in unexpected errors.
```

I have changed this expression slightly to avoid misunderstandings.